dotnet/src/AutoGen.Core/Orchestrator/RolePlayOrchestrator.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// RolePlayOrchestrator.cs
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AutoGen.Core;
+
+public class RolePlayOrchestrator : IOrchestrator
+{
+    private readonly IAgent admin;
+    private readonly Graph? workflow = null;
+    public RolePlayOrchestrator(IAgent admin, Graph? workflow = null)
+    {
+        this.admin = admin;
+        this.workflow = workflow;
+    }
+
+    public async Task<IAgent?> GetNextSpeakerAsync(
+        OrchestrationContext context,
+        CancellationToken cancellationToken = default)
+    {
+        var candidates = context.Candidates.ToList();
+
+        if (candidates.Count == 0)
+        {
+            return null;
+        }
+
+        if (candidates.Count == 1)
+        {
+            return candidates.First();
+        }
+
+        // if there's a workflow
+        // and the next available agent from the workflow is in the group chat
+        // then return the next agent from the workflow
+        if (this.workflow != null)
+        {
+            var lastMessage = context.ChatHistory.LastOrDefault();
+            if (lastMessage == null)
+            {
+                return null;
+            }
+            var currentSpeaker = candidates.First(candidates => candidates.Name == lastMessage.From);
+            var nextAgents = await this.workflow.TransitToNextAvailableAgentsAsync(currentSpeaker, context.ChatHistory);
```

Comment:
Nit : should `cancellationToken` be passed?

Reply:
#3111